### PR TITLE
Help Center: Add logic for archived conversations

### PR DESCRIPTION
**Pull request by agrullon95**
Fixes: https://github.com/Automattic/dotcom-forge/issues/9574

## Proposed Changes

| Recent | Archived |
| ------------- | -----------|
|CleanShot 2024-10-29 at 17 27 14@2x| CleanShot 2024-10-29 at 17 28 56@2x |

* Add logic for Help Center archived conversations.
* Refactor recent/archived conversations logic.
* Removes blue focus color on `Recent` and `Archived` tabs.
CleanShot 2024-10-29 at 17 16 41@2x

* Add `createdAt` to metadata when a new conversation is created with Smooch.
* Sort `Recent` conversations by unread messages + last message received. Unread conversations will be at the top of the recent conversations list.

## Why are these changes being made?

* Allow users to view their old conversations and create a separation between older conversations and recent conversations.

## Testing Instructions

* Checkout or use live link
* Go to `/home` with the `?flags=help-center-experience` flag enabled.
* Open the Help Center , click on the recent conversations section, and verify the list is populated correctly.
* To test the archived conversations, you can disable the flag, and modify the smooch.CreateConversation `metadata.createdAt` property ( https://github.com/Automattic/wp-calypso/pull/95839/files#diff-ae6e358ec645887d4afd9f77bb59e2d9aafbf1d71881dfbf823736d594d03b1bR36) to a timestamp from a year ago (https://www.timestamp-converter.com/ - use UTC timestamp) or you can create a new chat and hardcode the conversation `metadata.createdAt` property here: https://github.com/Automattic/wp-calypso/pull/95839/files#diff-3c5495b3c5142d5f12c2d9941b42c29042a6a7a79f64d6c812ab1ef600a45d5bR41.
* Test with a timestamp that is less than a year ago.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: packages/help-center/src/components/help-center-chat-history.scss [hunk 1]
```diff
@@ -9,6 +9,15 @@
 		padding: 16px;
 		font-size: 0.75rem;
 	}
+	
+	.components-tab-panel__tabs {
```
Comment (agrullon95) [inline packages/help-center/src/components/help-center-chat-history.scss]: Removes the blue focus color on the `Recent` and `Archived` tabs

File: packages/odie-client/src/query/use-create-zendesk-conversation.ts [hunk 2]
```diff
@@ -33,16 +33,18 @@ export const useCreateZendeskConversation = (): ( () => Promise< void > ) => {
...
-				setSupportProvider( 'zendesk' );
-				setChatStatus( 'loaded' );
-				setChat( ( prevChat ) => {
-					return {
-						...prevChat,
-						conversationId: conversation.id,
-					};
-				} );
-			} );
+			Smooch.createConversation( { metadata: { odieChatId: chatId, createdAt: Date.now() } } ).then(
```
Comment (agrullon95) [inline packages/odie-client/src/query/use-create-zendesk-conversation.ts:45]: Added a timestamp of the current time when a conversation is created.

File: packages/help-center/src/components/utils.tsx [hunk 3]
```diff
@@ -14,3 +15,45 @@ export const generateContactOnClickEvent = (
...
+} ) => {
+	const recentConversations: ZendeskConversation[] = [];
+	const archivedConversations: ZendeskConversation[] = [];
+
+	if ( ! conversations ) {
+		return {
+			recentConversations,
+			archivedConversations,
+		};
+	}
```
Comment (escapemanuele) [inline packages/help-center/src/components/utils.tsx]: Do we need this check? If conversations is null it is not gonna cycle in `conversations.forEach` and it is going to return anyways at the end of the function!

File: packages/help-center/src/components/utils.tsx [hunk 4]
```diff
@@ -14,3 +15,45 @@ export const generateContactOnClickEvent = (
...
+			const createdAtDate = new Date( createdAt as string | number | Date );
+			const now = new Date();
+			const oneYearAgo = new Date( now.setFullYear( now.getFullYear() - 1 ) );
+
+			if ( createdAtDate < oneYearAgo ) {
+				archivedConversations.push( conversation );
+			} else {
+				recentConversations.push( conversation );
+			}
+			return;
```
Comment (escapemanuele) [inline packages/help-center/src/components/utils.tsx]: Can you test if it is necessary to have this return?

File: packages/help-center/src/components/help-center-chat-history.scss [hunk 5]
```diff
@@ -9,6 +9,7 @@
 		padding: 16px;
 		font-size: 0.75rem;
 	}
+
```
Comment (escapemanuele) [inline packages/help-center/src/components/help-center-chat-history.scss]: No need to add this empty line

File: packages/help-center/src/components/utils.tsx (hunk 3 above)
Comment (agrullon95) [reply packages/help-center/src/components/utils.tsx]: This is not needed. I removed this condition.

File: packages/help-center/src/components/utils.tsx (hunk 4 above)
Comment (agrullon95) [reply packages/help-center/src/components/utils.tsx]: This is not needed as well. I removed the return statement.

File: packages/help-center/src/components/help-center-chat-history.scss (hunk 5 above)
Comment (agrullon95) [reply packages/help-center/src/components/help-center-chat-history.scss]: I removed this empty line/tab.

**Comment by agrullon95** [comment]
@escapemanuele - Thanks for the feedback. I addressed your comments.

File: packages/help-center/src/components/help-center-chat-history.tsx [hunk 6]
```diff
@@ -9,10 +9,40 @@ import NavItem from 'calypso/components/section-nav/item';
...
+import { getFilteredConversations } from './utils';
 import type { ZendeskConversation } from '@automattic/odie-client';
 
 import './help-center-chat-history.scss';
 
+const Conversations = ( { conversations }: { conversations: ZendeskConversation[] } ) => {
+	const { __ } = useI18n();
+	if ( ! conversations || ! conversations.length ) {
+		return <div className="help-center-chat-history__no-results">{ __( 'Nothing found…' ) }</div>;
+	}
```
Comment (escapemanuele) [inline packages/help-center/src/components/help-center-chat-history.tsx:21]: I'd possibly not show anything if we don't have old chats.
Wdyt @lucasmendes-design?

image

**Comment by renancarvalho** [comment]
I was able to see the archived chat when I manipulated the timestamp

image


Besides the blue color, it LGTM 👍

File: packages/help-center/src/components/utils.tsx [hunk 7]
```diff
@@ -14,3 +15,37 @@ export const generateContactOnClickEvent = (
 		} );
 	}
 };
+
+export const getFilteredConversations = ( {
```
Comment (renancarvalho) [inline packages/help-center/src/components/utils.tsx:25]: Not mandatory for this PR, I think this would be nicer as a hook, wdyt?

**Review by renancarvalho — APPROVED** [review]

**Comment by lucasmendes-design** [comment]
Not sure if this is related to this but when I tested with the flag `?flags=help-center-experience` the hover state is strange. 

image

These tabs should behave like the one below. Example: 

Normal state
image

Hover state
image

**Comment by agrullon95** [comment]
> Not sure if this is related to this but when I tested with the flag ?flags=help-center-experience the hover state is strange.

@lucasmendes-design - Thats unrelated but we can definitely fix it in another PR.

**Comment by lucasmendes-design** [comment]
all good :)

**Pull request merged**